src/main/java/aquality/selenium/core/elements/DesiredState.java

```diff
@@ -0,0 +1,47 @@
+package aquality.selenium.core.elements;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.function.Predicate;
+
+/**
+ * Defines desired state for element with ability to handle exceptions.
+ */
+public class DesiredState {
+
+    private final Predicate<WebElement> desiredStatePredicate;
+    private final String stateName;
```

Comment:
Just to clarify, why should we use String, but not ElementState

Reply:
because it is not related to ElementState. DesiredState could be initialized with any Predicate<WebElement>, and we need the stateName for logging, e.g. "enabled", "clickable" and so on